<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="4" passed="2" failed="1" skipped="1">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-07-22T23:00:51 IST" name="Automation Suite (ExtentReports for Reporting with ITestListeners)" finished-at="2025-07-22T23:01:09 IST" duration-ms="18015">
    <groups>
      <group name="Regression">
        <method signature="LoginTest.testLoginWithLockedCredentials()[pri:2, instance:tests.LoginTest@6cc558c6]" name="testLoginWithLockedCredentials" class="tests.LoginTest"/>
        <method signature="LoginTest.testLoginWithInvalidCredentials()[pri:3, instance:tests.LoginTest@6cc558c6]" name="testLoginWithInvalidCredentials" class="tests.LoginTest"/>
      </group> <!-- Regression -->
      <group name="Sanity, Regression">
        <method signature="LoginTest.testLoginWithValidCredentials()[pri:1, instance:tests.LoginTest@6cc558c6]" name="testLoginWithValidCredentials" class="tests.LoginTest"/>
        <method signature="LogoutTest.testLogout()[pri:1, instance:tests.LogoutTest@61544ae6]" name="testLogout" class="tests.LogoutTest"/>
      </group> <!-- Sanity, Regression -->
    </groups>
    <test started-at="2025-07-22T23:00:51 IST" name="Functional Test" finished-at="2025-07-22T23:01:09 IST" duration-ms="18015">
      <class name="tests.LoginTest">
        <test-method is-config="true" signature="setup()[pri:0, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:00:51 IST" name="setup" finished-at="2025-07-22T23:00:54 IST" duration-ms="2759" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testLoginWithValidCredentials()[pri:1, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:00:54 IST" name="testLoginWithValidCredentials" description="Test to verify login with valid credentials." finished-at="2025-07-22T23:00:55 IST" duration-ms="942" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLoginWithValidCredentials -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:00:55 IST" name="tearDown" finished-at="2025-07-22T23:00:57 IST" duration-ms="1620" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:00:57 IST" name="setup" finished-at="2025-07-22T23:00:59 IST" duration-ms="2786" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testLoginWithLockedCredentials()[pri:2, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:00:59 IST" name="testLoginWithLockedCredentials" description="Test to verify login with locked credentials." finished-at="2025-07-22T23:00:59 IST" duration-ms="5" status="SKIP">
          <exception class="org.testng.SkipException">
            <message>
              <![CDATA[This test is Intentionally skipped for ITestListener test.]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.SkipException: This test is Intentionally skipped for ITestListener test.
at tests.LoginTest.testLoginWithLockedCredentials(LoginTest.java:26)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.SkipException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLoginWithLockedCredentials -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:00:59 IST" name="tearDown" finished-at="2025-07-22T23:01:02 IST" duration-ms="2462" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:01:02 IST" name="setup" finished-at="2025-07-22T23:01:04 IST" duration-ms="2093" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testLoginWithInvalidCredentials()[pri:3, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:01:04 IST" name="testLoginWithInvalidCredentials" description="Test to verify login with invalid credentials." finished-at="2025-07-22T23:01:05 IST" duration-ms="580" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error message on the Login Page is incorrect. expected [Username Required] but found [Epic sadface: Username is required]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error message on the Login Page is incorrect. expected [Username Required] but found [Epic sadface: Username is required]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at tests.LoginTest.testLoginWithInvalidCredentials(LoginTest.java:34)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLoginWithInvalidCredentials -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.LoginTest@6cc558c6]" started-at="2025-07-22T23:01:05 IST" name="tearDown" finished-at="2025-07-22T23:01:05 IST" duration-ms="285" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.LoginTest -->
      <class name="tests.LogoutTest">
        <test-method is-config="true" signature="setup()[pri:0, instance:tests.LogoutTest@61544ae6]" started-at="2025-07-22T23:01:05 IST" name="setup" finished-at="2025-07-22T23:01:07 IST" duration-ms="1907" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testLogout()[pri:1, instance:tests.LogoutTest@61544ae6]" started-at="2025-07-22T23:01:07 IST" name="testLogout" description="Test to verify logout functionality." finished-at="2025-07-22T23:01:08 IST" duration-ms="1460" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLogout -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.LogoutTest@61544ae6]" started-at="2025-07-22T23:01:08 IST" name="tearDown" finished-at="2025-07-22T23:01:09 IST" duration-ms="199" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.LogoutTest -->
    </test> <!-- Functional Test -->
  </suite> <!-- Automation Suite (ExtentReports for Reporting with ITestListeners) -->
</testng-results>
